### DISCUSSION TITLE
Import navigation tracks

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#141] [Q&A] [answered]
V 3.018 (Test by desktop)

1. - The import menu Routing set is not respected @ the import of a navtrack+wpt.
2. - The (red) track overlay color set is not memorised in the next sessions.

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the suggestion.

1. Navigation tracks are automatically imported as navigation tracks,
this was the decision in the discussion before the implementation.
(avoid extra menu to select normal or navigation type of gpx)

2. Focus and available resources are on Android development. :slightly_smiling_face: 
So not all Android features can be implemented on the desktop.

**Reply by 0709wiwiwi** [reply]
> 1. Navigation tracks are automatically imported as navigation tracks

Correct.  But only If set GPX > **Automatic**

**Reply by devemux86** [reply]
> Correct. But only If set GPX > Automatic !

It was never decided or designed like that.
The detection of navigation vs normal gpx happens automatically, without user intervention.

What you describe is a new suggestion for how the import could work. :slightly_smiling_face:

**Reply by devemux86** [reply]
The list (Automatic, Track, Route, Waypoints) is useful with the **normal gpx**,
because import of a normal gpx means **route calculation** with the gpx data.

**Navigation gpx** are supposed to contain **offline routes** for navigation (not extra wpt).
Import of a navigation gpx is **offline without route calculation** (like kurviger import).

I will think about your suggestion and if we can enhance the import process.

**Reply by 0709wiwiwi** [reply]
**You are right** 
_It's just a coincidence that I noticed this.

**Reply by devemux86** [reply]
> It's just a coincidence that I noticed this.

I like these tests that push the software to its limits! :+1: 

> How to avoid an entire navigation track recalculation result if you change something, by using a "strange" router, between two via or shaping points. The total track can thus be changed (too) drastically or the recalculation is even impossible due to different routing rules.

This is another new feature already planned for Cruiser:
- https://github.com/devemux86/cruiser/discussions/142

**Comment by devemux86** [comment]
Also, there is currently no support of gpx wpt in the navigation tracks.
Navigation tracks are now meant to keep routes for offline navigation.

For example, Cruiser currently fills the gpx **trk** in navigation track export (not gpx wpt).
Support of gpx wpt in navigation tracks is a different feature we could see in the future.

**Comment by devemux86** [comment]
> Correct. But only If set GPX > Automatic !
Not If SET: Routing by Track or Route or Waypoints.

It will be available in the next Cruiser update.

**Comment by devemux86** [comment, marked as answer]
Implemented in [Cruiser 3.0.19](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
